x-pack/plugins/actions/server/builtin_action_types/servicenow/api.ts

```diff
@@ -3,5 +3,138 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { flow } from 'lodash';
+import {
+  ExternalServiceParams,
+  PushToServiceApiHandlerArgs,
+  HandshakeApiHandlerArgs,
+  GetIncidentApiHandlerArgs,
+  ExternalServiceApi,
+} from './types';
 
-export { api } from '../case/api';
+// TODO: to remove, need to support Case
+import { transformers, Transformer } from '../case/transformers';
+import { PushToServiceResponse, TransformFieldsArgs } from './case_types';
+import { prepareFieldsForTransformation } from '../case/utils';
+
+const handshakeHandler = async ({
+  externalService,
+  mapping,
+  params,
+}: HandshakeApiHandlerArgs) => {};
+const getIncidentHandler = async ({
+  externalService,
+  mapping,
+  params,
+}: GetIncidentApiHandlerArgs) => {};
+
+const pushToServiceHandler = async ({
+  externalService,
+  mapping,
+  params,
+  secrets,
+}: PushToServiceApiHandlerArgs): Promise<PushToServiceResponse> => {
+  const { externalId, comments } = params;
+  const updateIncident = externalId ? true : false;
+  const defaultPipes = updateIncident ? ['informationUpdated'] : ['informationCreated'];
+  let currentIncident: ExternalServiceParams | undefined;
+  let res: PushToServiceResponse;
+
+  if (externalId) {
+    currentIncident = await externalService.getIncident(externalId);
+  }
+
+  let incident = {};
+  // TODO: should be removed later but currently keep it for the Case implementation support
+  if (mapping) {
+    const fields = prepareFieldsForTransformation({
+      externalCase: params.externalObject,
+      mapping,
+      defaultPipes,
+    });
+
+    incident = transformFields({
+      params,
+      fields,
+      currentIncident,
+    });
+  } else {
+    incident = { ...params, short_description: params.title, comments: params.comment };
+  }
+
+  if (updateIncident) {
+    res = await externalService.updateIncident({
+      incidentId: externalId,
+      incident,
+    });
+  } else {
+    res = await externalService.createIncident({
+      incident: {
+        ...incident,
+        caller_id: secrets.username,
+      },
+    });
+  }
+
+  // TODO: should temporary keep comments for a Case usage
+  if (
+    comments &&
+    Array.isArray(comments) &&
+    comments.length > 0 &&
+    mapping &&
+    mapping.get('comments')?.actionType !== 'nothing'
+  ) {
+    res.comments = [];
+
+    const fieldsKey = mapping.get('comments')?.target ?? 'comments';
+    for (const currentComment of comments) {
```

Comment:
Yeah, good point. Unfortunately I didn't find a way how to do a multiple updates using a single request in ServiceNow REST API. But I don't think that it is a big issue, because currently we don't have such a use case when number of comments will be more than a few.

Reply:
I also did not find a way of how to do multiple updates in ServiceNow REST API.